GSG_Cypress/cypress/e2e/OrangeHRM/leaveRequest.spec.cy.ts

```diff
@@ -0,0 +1,127 @@
+          expect(response).property('status').to.eq(200);
+        });
+      }).then(() => {
+        cy.request({
+          method: 'POST',
+          url: 'https://opensource-demo.orangehrmlive.com/web/index.php/api/v2/leave/leave-entitlements',
+          body: {
+            empNumber: empNumberResponse,
+            leaveTypeId: 9,
+            fromDate: "2023-01-01",
+            toDate: "2024-08-31",
+            entitlement: "20"
+          }
+        }).then((response) => {
+          expect(response).property('status').to.eq(200);
+        });
+      });
+    })
+  });
+
+
+
+  it("logout and then login", () => {
+    logout();
+    login(username, password);
+    cy.get('@leaveData').then((infoData: any) => {
+      requestLeave(infoData.requestLeave.comment,
+        infoData.requestLeave.duration,
+        infoData.requestLeave.fromDate,
+        infoData.requestLeave.leaveTypeId,
+        infoData.requestLeave.partialOption,
+        infoData.requestLeave.toDate);
+
+      cy.then(() => {
+        logout();
+        login("Admin", "admin123");
+        approveReject();
+        logout();
+        login(username, password);
+        cy.visit("https://opensource-demo.orangehrmlive.com/web/index.php/leave/viewMyLeaveList")
+        cy.get('@data').then((infoData1: any) => {
+          checkDataInTable('.oxd-table', ["", `${infoData.requestLeave.fromDate} to ${infoData.requestLeave.toDate}`, `${infoData1.addEmployee.firstName} ${infoData1.addEmployee.middleName} ${infoData1.addEmployee.lastName}`]);
+
+        })
+      });
+      logout();
+      login("Admin", "admin123");
+
+      cy.request({
+        method: 'POST',
+        url: 'https://opensource-demo.orangehrmlive.com/web/index.php/api/v2/recruitment/vacancies',
+        body: {
+          "name": "vacancy",
+          "jobTitleId": 10,
+          "employeeId": empNumberResponse,
+          "numOfPositions": null,
+          "description": "",
+          "status": true,
+          "isPublished": true
+        }
+      }).then((response) => {
+        expect(response).property('status').to.equal(200)
+        vacancyId = response.body.data.id;
+        console.log(vacancyId)
+      }).then(() => {
+        cy.visit(`https://opensource-demo.orangehrmlive.com/web/index.php/recruitment/addJobVacancy/${vacancyId}`)
+        cy.get('.orangehrm-header-container  .oxd-button', { timeout: 30000 }).click({ force: true }).then(() => {
+          cy.get('input[type=file]').selectFile("cypress/fixtures/file.txt", { force: true }).then(() => {
+            cy.get('.oxd-table-loader > .oxd-loading-spinner-container').should("exist").then(() => {
+
+              cy.get('.oxd-table-loader > .oxd-loading-spinner-container').should("not.exist").then(() => {
+
+
+                cy.get('.oxd-form  .oxd-form-actions  .oxd-button--secondary').last().click({ force: true })
+                cy.get('.oxd-table-card  .oxd-table-row', { timeout: 30000 }).contains('file.txt')
+              })
+
```

Comment:
**Why you declare every request and action in the test ?!,This way is not acceptable** 
Should declare method for every request and for every action and for every assertion